### DISCUSSION TITLE
Migration from axios-cache-adapter

### DISCUSSION THREAD

**Discussion by Tam2** [arthurfiorette/axios-cache-interceptor#752] [Q&A] [answered]
Hello,

We're looking to migrate from axios-cache-adapter

I have a few queries:

- How do we set up a default cache maxAge? - Would like to set this to 15mins by default, but will use the ttl on individual requests if i need a different cache time for any particular requests
- How can i set default headers on all requests? i need to add some headers to all of my requests automatically
- I see there is an `await axios.storage.remove("key")` function, how can i remove all cached entries? i can't see a `axios.storage.removeAll()` or can we pass in with an empty key?

**Comment by arthurfiorette** [comment]
Hey @Tam2, which parts of the [`axios-cache-interceptor`](https://axios-cache-interceptor.js.org/guide/getting-started) documentation did you not understand?

**Reply by Tam2** [reply]
@arthurfiorette - The parts where it doesn't answer the questions I've asked

1.  Wasn't clear on the first read of https://axios-cache-interceptor.js.org/config that request specific config can be used globally, only noticed the 'TIP' box second time around 
2. Still doesn't state if we need to do anything different to pass in headers, will assume your plugin returns an instance of Axios so we can do this - https://axios-http.com/docs/config_defaults 
3. There is nothing in your documentation which states how to remove all entries in the cache

**Reply by arthurfiorette** [reply, marked as answer]
2 -> It returns the same axios instance, but with updated typings, you can just install this library and wait for you LSP add completions to all available options.

3 -> It depends on your storage implementation, we do not support it by default buy you can extend any storage with https://axios-cache-interceptor.js.org/guide/storages#buildstorage.

**Reply by arthurfiorette** [reply]
For example, if you are using localStorage, you can just list all of its keys and delete everyone that starts with the provided prefix and delete them.

https://axios-cache-interceptor.js.org/guide/storages#buildstorage

**Reply by Tam2** [reply]
It looks like for localStorage, the data is cached in `storage.data`, not sure if we have access to this? as we could reset this by setting it to an empty object

**Reply by arthurfiorette** [reply]
For memory storage it is, you can lookup the actual implementation here: https://github.com/arthurfiorette/axios-cache-interceptor/blob/main/src/storage/memory.ts

`data` property is available, you can get type autocompletion by casting the `AxiosStorage` type to `MemoryStorage`

**Reply by Tam2** [reply]
Great, i'm setting `axios.storage.data = Object.create(null);` which seems to do the trick of emptying the cache

**Comment by chiptus** [comment]
@Tam2 @arthurfiorette what did you do with maxAge?

**Reply by Tam2** [reply]
I used the ttl

e.g.

```
const api = setupCache(instance, {
  ttl: 1000 * 60 * 30,
});
```
